Add callback_url/acct information for Sidekiq PuSH workers Exception.

clworld opened the pull request:
Some exception does not report hostname in exception message.
So, hostname does not shown in Sidekiq error console.
You need check database tables to know which host returns error.
With this PR, you can easily see which server is in trouble...
![sidekiq_fix3](https://user-images.githubusercontent.com/1680550/28429873-4f25b35e-6db9-11e7-9a38-c12ce9d00980.PNG)

beatrix-bitrot commented:
ok maybe fix the broken tests d:

still... i really want this

nightpool commented on app/lib/exceptions.rb:
```diff
@@ -8,11 +8,11 @@ class RaceConditionError < Error; end
-    def to_s
-      "#{@response.uri} returned code #{@response.code}"
+      if response.respond_to? :url
+        super("#{response.uri} returned code #{response.code}")
```
url vs uri?

clworld replied:
Oh. mistake…

nightpool commented:
What is the difference between "response.uri" and "acct"? One gives more meaningful information, in my opinion.

nightpool commented:
This feels like it's redundant now that we use MastodonUnexpectedResponse.

clworld commented:
I'm confused about RSpec executes Sidekiq worker directly and exceptions thrown by workers returns to caller...
I choose way that don't change Exception class. but this way assumes all StandardError are reproducible..
Another way is wrap errors by some Exception (Mastodon::WorkerError). It breaks rspec which checks which exception is raised.

clworld commented:
@nightpool 
responce.uri is what UnexpectedResponseError show on .message.
I don't change UnexpectedResponseError's output. I changed UnexpectedResponseError to make it reproducible.

What I changed in this PR is add acct/callback_url to Exceptions which Sidekiq workers raises.
callback_url is url to post OStatus payload.
acct is Mastodon account name to subscribe. For example: "clworld@md.ggtea.org" is my account's acct.

nightpool commented:
okay, that makes sense. LGTM

clworld commented:
I send alternative implementation: https://github.com/tootsuite/mastodon/pull/4283
which don't use 'raise e.class "..."' thing. but wrap all exceptions.
